cli/arg/arg.go

```diff
@@ -141,6 +156,23 @@ func GetBazelArgs(args []string) []string {
 	return append([]string{}, args[:splitIndex]...)
 }
 
+// Splits out flags that affect the bb cli executable. These should be set before
+// the bb cli command.
+// Ex. `bb <--cli_args> execute`, bb <--cli_args> remote build //...
+// Returns cli args and the original args with the cli args removed
+func SplitCLIArgs(args []string) ([]string, []string) {
+	splitIndex := 0
+	for splitIndex < len(args) {
+		if !strings.HasPrefix(args[splitIndex], "--") {
+			break
+		}
+		splitIndex++
+	}
+	cliArgs := args[:splitIndex]
+	remainingArgs := args[splitIndex:]
+	return cliArgs, remainingArgs
+}
```

Comment:
I think this logic means that the `bb` flags have to take the `--arg=value` form, which might be confusing because Bazel supports passing args like `--arg value` (without the `=`).

Go's `flag.Parse` should stop parsing as soon as it sees the first non-flag arg, which is what we want in this case I think. Would it work to just use `flag.Parse` and then parse the remaining non-flag args using our existing parsing logic?

For example:

```
var verbose = flag.Bool(...)
os.Args = []string{"bb", "--target", "remote.buildbuddy.dev", "remote", "build" "--foo=bar"}
flag.Parse()
// Should stop parsing at `remote`
flag.Args() // should return []string{"remote" "build", "--foo=bar"}
```